articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,110 @@
+  - login
+  - user profile
+  - logout
+  - nodejs
+contentType: tutorial
+useCase: quickstart
+github:
+  path: 01-Login
+---
+
+<%= include('../../../_includes/_new_app', { hideDownloadSample: true, hideDashboardScreenshot: true }) %>
+<%= include('../../../_includes/_callback_url') %>
+<%= include('../../../_includes/_logout_url' }) %>
+
+## Integrate Auth0
+### Install Dependencies
+Install the following dependencies
+
+- [`express-session`](https://github.com/auth0/express-session) - simple session middleware for Express (any session middleware for express will work).
+- [`express-openid-connect`](https://github.com/auth0/express-openid-connect) - an [OIDC](/protocols/oidc) compliant authentication library.
+
+```sh
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+### Handling server responses
+Your application will need to parse url-encoded payloads sent back from the Auth0 server.  Express provides a middleware for this `express.urlencoded`. You might already be using `body-parser`, this will work too.
+
+```js
+const express = require('express');
+const app = express();
+
+app.use(express.urlencoded({
+  extended: false
+}));
+```
+
+### User sessions
+In order to keep the user logged in across requests you will need to use session middleware.  The session will allow you to recognize the user on concurrent requests. In this example, you will use `express-session` which stores the session in memory.
+
+```js
+const session = require('express-session');
+
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+```
+
+### Authentication
+The last part of integrating Auth0 into your application will be using the `auth` middleware supplied to you from the Auth0 library `express-opeinid-connect`.  View the official [Api Documentation](https://github.com/auth0/express-openid-connect/blob/4374c0502d4aedf8b3975e7c5f26929b305b32f6/API.md) for configuration options and information.
+
+```js
+const { auth } = require('express-openid-connect');
+
+app.use(auth({
+  required: false,
+  auth0Logout: true,
+  baseURL: 'http://localhost:3000',
+  issuerBaseURL: 'https://${account.namespace}',
+  clientID: '${account.clientId}'
+}));
+```
+
+## Protect a Route
+Now that your application has the required middleware to integrate Auth0 in your application, you can start protecting routes.
+
+Add the `requiresAuth` middleware supplied from `express-openid-connect` for routes that require authentication.  Any route using this middleware will check for a valid user session and, if one does not exist, it will redirect the user to log in.
+
+```js
+const { requiresAuth } = require('express-openid-connect');
+
+app.get('/profile', requiresAuth(), (req, res) => {
+  // when a user is logged in you can access the user object via `req.openid.user`
+  res.send(JSON.stringify(req.openid.user));
+});
+```
+
+## Login and Logout
```

Comment:
Maybe instead of a main section header for both, Login and Logout get their own sections.

Reply:
pretty small sections, but we let's test it